interfaces/common/Common.h

```diff
@@ -26,4 +29,19 @@ U align(T number, U alignment) {
 }
 }
 
-#endif
\ No newline at end of file
+inline int getProcessId() {
+  std::vector<std::string> rankEnvVars{{"SLURM_PROCID"},
```

Comment:
what exactly is the difference to the MPI rank?

Reply:
renamed to `getMpiRankFromEnv`. Note, we need MPI rank only for outputting some debug/info to the console.

The device instance should be initialized before a call to MPI_Init (it used to be a requirement for some GPU-Aware MPI implementation). The safest option would be reading the rank from the env. variable.